src/widgets/merch-catalog/ui/merch-filter/filtered-catalog/filtered-catalog.tsx

```diff
@@ -0,0 +1,147 @@
+        params.set('search', currentSearchTerm);
+      }
+      currentSelectedTypes.forEach((type) => {
+        params.append('type', type);
+      });
+
+      const queryString = params.toString();
+
+      router.replace(queryString ? `${pathname}?${queryString}` : pathname, { scroll: false });
+    },
+    [pathname, router],
+  );
+
+  useEffect(() => {
+    if (initialProducts && initialProducts.length > 0) {
+      const uniqueTags = Array.from(
+        new Set(initialProducts.flatMap((product) => product.tags || []).filter((tag) => tag)),
+      ).sort();
+
+      setAllAvailableTags(uniqueTags);
+    } else {
+      setAllAvailableTags([]);
+    }
+  }, [initialProducts]);
+
+  const filteredProducts = useMemo(() => {
+    let productsToFilter = initialProducts || [];
+
+    if (!Array.isArray(productsToFilter)) {
+      productsToFilter = [];
+    }
+
+    if (searchTerm.trim() !== '') {
+      productsToFilter = productsToFilter.filter((product) =>
+        product.title.toLowerCase().includes(searchTerm.toLowerCase()),
+      );
+    }
+
+    if (selectedTypes.length > 0) {
+      productsToFilter = productsToFilter.filter((product) =>
+        selectedTypes.some((typeValue) => (product.tags || []).includes(typeValue)),
+      );
+    }
+    return productsToFilter;
+  }, [initialProducts, searchTerm, selectedTypes]);
+
+  const handleSearchChange = (newSearchTerm: string) => {
+    setSearchTerm(newSearchTerm);
+    updateUrl(newSearchTerm, selectedTypes);
+  };
+
+  const handleTypeChange = (typeValue: string) => {
+    const newSelectedTypes = selectedTypes.includes(typeValue)
+      ? selectedTypes.filter((t) => t !== typeValue)
+      : [...selectedTypes, typeValue];
+
+    setSelectedTypes(newSelectedTypes);
+    updateUrl(searchTerm, newSelectedTypes);
+  };
+
+  const handleClearFilters = () => {
+    setSearchTerm('');
+    setSelectedTypes([]);
+    updateUrl('', []);
+  };
+
+  const toggleMobileFiltersExpansion = () => {
+    setAreMobileFiltersExpanded(!areMobileFiltersExpanded);
+  };
+
+  useEffect(() => {
+    if (!isMobileLayout && areMobileFiltersExpanded) {
+      setAreMobileFiltersExpanded(false);
+    }
+  }, [isMobileLayout, areMobileFiltersExpanded]);
+
+  const hasActiveFilters = searchTerm.trim() !== '' || selectedTypes.length > 0;
+
+  return (
+    <div className={cx('filter-catalog')}>
+      {' '}
+      <div className={cx('filter-sidebar')}>
+        <FilterControls
+          allAvailableTags={allAvailableTags}
+          searchTerm={searchTerm}
+          selectedTags={selectedTypes}
+          hasActiveFilters={hasActiveFilters}
+          onSearchChange={handleSearchChange}
+          onTagChange={handleTypeChange}
+          onClearFilters={handleClearFilters}
+          isMobileLayout={isMobileLayout}
+          areTagsExpandedMobile={areMobileFiltersExpanded}
+          onToggleTagsExpansionMobile={toggleMobileFiltersExpansion}
+        />
+      </div>
+      <main>
```

Comment:
why main-tag here? all content between header and footer have to be in main, isn't it?